src/pip/_internal/download.py

```diff
@@ -936,18 +936,20 @@ def unpack_file_url(
     link_path = url_to_path(link.url_without_fragment)
     # If it's a url to a local directory
     if is_dir_url(link):
+
+        def ignore(d, names):
+            # Pulling in those directories can potentially
+            # be very slow.
+            # see discussion at:
+            #   https://github.com/pypa/pip/pull/6770
+            return ['.tox', '.nox'] if d == link_path else []
```

Comment:
Technically [the docs](https://docs.python.org/3/library/shutil.html#shutil.copytree) say

> i.e. a subset of the items in its second argument

Is it worth only returning these if they actually exist in `names`? Per the tests it does work as-is.

Reply:
I think it's not worth it but I will fix if you insist.

Reply:
Not insisting, fine as-is for me. :+1: